src/App.js

```diff
@@ -21,45 +23,106 @@ const generateSquares = () => {
       currentId += 1;
     }
   }
-
+  // console.log(squares);
   return squares;
 };
 
 const App = () => {
   // This starts state off as a 2D array of JS objects with
   // empty value and unique ids.
   const [squares, setSquares] = useState(generateSquares());
+  const [player, setPlayer] = useState(playerOne);
+  const [isGameOver, setGameStatus] = useState(false);
+  const [winner, setWinner] = useState(winnerInitial);
 
-  // Wave 2
-  // You will need to create a method to change the square
-  //   When it is clicked on.
-  //   Then pass it into the squares as a callback
+  const updateOnClick = (updatedSquare) => {
+    const squaresData = squares.map((row) => {
+      row.map((square) => {
+        if (square.id === updatedSquare.id) {
+          if (square.value === '') {
+            square.value = player;
+            const updatedWinner = checkForWinner();
```

Comment:
As written, checkForWinner should not be called during the click handling. It is written to look at the squares piece of state, but if we are properly calculating a new state and setting it, then the updated squares state wouldn't become observable until the next render. It is working here, since the changes to the board values are cross-contaminating the current React squares without React knowing about it (due to JS storing references to data).

Instead, with how checkForWinner is written, it should be called in the body of the App component function, which would check for a winner on each re-render. Notice this would allow us to avoid introducing a piece of state to track the winner, or whether the game is over.